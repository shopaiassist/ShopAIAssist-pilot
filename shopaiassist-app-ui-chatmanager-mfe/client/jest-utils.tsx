import React, { JSXElementConstructor, ReactElement } from 'react';
import { Provider } from 'react-redux';
import { MemoryRouterProps } from 'react-router';
import { MemoryRouter } from 'react-router-dom';
import { I18nextProvider } from 'react-i18next';
import { merge } from 'ts-deepmerge';
import { configureStore, EnhancedStore } from '@reduxjs/toolkit';
import { ByRoleMatcher, render, RenderResult } from '@testing-library/react';
import { reducer, RootState, store as baseStore } from './src/store/store';
import i18nInstance from './src/utils/i18n';

type InitialEntries = MemoryRouterProps['initialEntries'];
type DeepPartial<T> = {
  [P in keyof T]?: T[P] extends object ? DeepPartial<T[P]> : T[P];
};
type PartialRootState = DeepPartial<RootState>;

interface RenderOptions {
  entries?: InitialEntries;
  state?: PartialRootState;
  store?: EnhancedStore;
}

interface SnapshotOptions {
  isModal?: boolean | ByRoleMatcher;
  state?: PartialRootState;
}

interface WrapperProps {
  children: ReactElement;
}

// export const mockApi = (apiInstance: AxiosInstance) => ({
//   delete: jest.spyOn(apiInstance, 'delete'),
//   get: jest.spyOn(apiInstance, 'get'),
//   post: jest.spyOn(apiInstance, 'post'),
//   put: jest.spyOn(apiInstance, 'put'),
//   patch: jest.spyOn(apiInstance, 'patch'),
// });

/** Redux store with optional default content */
export const mockStore = (state: PartialRootState = {}): EnhancedStore =>
  configureStore({ preloadedState: merge(baseStore.getState(), state), reducer: reducer as never });

/** Render function with access to override store/state */
export const renderAll = (ui: ReactElement, { entries, state = {}, store }: RenderOptions = {}): RenderResult => {
  const data = store || mockStore(state);
  const Wrapper = ({ children }: WrapperProps) => (
    <MemoryRouter initialEntries={entries}>
      <Provider store={data}>
        <I18nextProvider i18n={i18nInstance}>{children}</I18nextProvider>
      </Provider>
    </MemoryRouter>
  );
  return render(ui, { wrapper: Wrapper as JSXElementConstructor<{ children: React.ReactNode }> });
};

/** Render function with i18n provider */
export const renderWithI18n = (ui: ReactElement): RenderResult => {
  const I18nWrapper = ({ children }: WrapperProps) => <I18nextProvider i18n={i18nInstance}>{children}</I18nextProvider>;
  return render(ui, { wrapper: I18nWrapper as JSXElementConstructor<{ children: React.ReactNode }> });
};

/** Render function with access to override store/state */
export const renderWithRedux = (ui: ReactElement, { state = {}, store }: RenderOptions = {}): RenderResult => {
  const data = store || mockStore(state);
  const ReduxWrapper = ({ children }: WrapperProps) => (
    <Provider store={data}>
      <I18nextProvider i18n={i18nInstance}>{children}</I18nextProvider>
    </Provider>
  );
  return render(ui, { wrapper: ReduxWrapper as JSXElementConstructor<{ children: React.ReactNode }> });
};

/** Get content to test against a snapshot */
export const getSnapshot = (ui: ReactElement, state?: PartialRootState): ChildNode =>
  renderAll(ui, { state }).container.firstChild as ChildNode;

/** Get modal content to test against a snapshot */
export const getSnapshotModal = (ui: ReactElement, state?: PartialRootState, role?: ByRoleMatcher): ChildNode =>
  renderAll(ui, { state }).getByRole(role || 'presentation');

/** Helper function to generate snapshots */
export const snapshot = (ui: ReactElement, { isModal, state }: SnapshotOptions = {}) => {
  const role = typeof isModal === 'boolean' ? undefined : isModal;
  // eslint-disable-next-line no-extra-boolean-cast
  const snapshot = !!isModal ? getSnapshotModal(ui, state, role) : getSnapshot(ui, state);
  expect(snapshot).toMatchSnapshot();
};

// re-export everything
export * from '@testing-library/react';
