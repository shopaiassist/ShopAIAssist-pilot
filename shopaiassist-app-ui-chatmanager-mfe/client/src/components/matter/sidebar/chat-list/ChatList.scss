@import "../../../../utils/styles/imports";

.chat-list-container {
   display: flex;
   flex-direction: column;
   width: 100%;
   overflow-y: auto;
   overflow-x: hidden;
   direction: rtl;

   .chat-list {
      direction: ltr;

      .timeframe-text {
         color: $saf-color-gray-400;
         font-size: $saf-font-size-sm;
         font-weight: 600;
         margin: 0.75rem;
      }
   }
}

.empty-chat-list {
   color: $saf-color-interactive-read-only-strong;
   font: $saf-type-body-default-md-regular-standard;
   display: flex;
   align-self:flex-end;
   margin-left: 1.5rem;
}
