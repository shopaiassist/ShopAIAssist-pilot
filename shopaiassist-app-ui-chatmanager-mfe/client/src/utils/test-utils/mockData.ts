import { TreeItem, FolderItem } from '../../@types/sidebar';

export const mockTree: (TreeItem | FolderItem)[] = [
  {
    treeItemId: '9a4144e0-65ea-4f6c-8cd5-43a0a6d1e634',
    name: 'Test Folder For Created At 2',
    type: 'folder',
    createdAt: new Date('2024-05-06T15:26:57.992Z'),
    updatedAt: new Date('2024-05-06T15:26:57.992Z'),
    fileCollectionId: 'whatever'
  },
  {
    treeItemId: '5aa13d2b-3f04-4a91-a84b-bf214420b47c',
    name: 'Test Folder For Created At 2',
    type: 'folder',
    createdAt: new Date('2024-05-06T15:19:07.430Z'),
    updatedAt: new Date('2024-05-06T15:19:07.430Z'),
    fileCollectionId: 'whatever'
  },
  {
    type: 'folder',
    name: 'New Folder 100',
    description: 'This folder is used for whatever, its created via postman.',
    matterId: 'Whatever Folder 100',
    treeItemId: '4c484fb3-faf6-4fe8-9930-410772f141d7',
    createdAt: new Date('2024-04-25T19:21:07.787Z'),
    updatedAt: new Date('2024-04-25T19:21:07.787Z'),
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-03-01T17:00:33.564Z'),
    treeItemId: 'e35607c7-be8f-4172-a3cd-ae7e39bc2b96',
    name: 'A chat named desire',
    type: 'chat',
    updatedAt: new Date('2024-05-06T18:09:49.245Z'),
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-23T16:30:33.080Z'),
    updatedAt: new Date('2024-02-23T16:30:33.080Z'),
    treeItemId: '8c1b78b3-a798-4557-871a-bff6c8ac804e',
    name: 'new2',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-23T16:30:17.078Z'),
    updatedAt: new Date('2024-02-23T16:30:17.078Z'),
    treeItemId: 'b39f37a3-6781-4445-8573-77658d9ee7ae',
    name: 'new1',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-23T16:26:17.254Z'),
    updatedAt: new Date('2024-02-23T16:26:17.254Z'),
    treeItemId: '9bc3c583-a6e4-41cc-92f1-27e1be3ba4f6',
    name: 'newnewenewnew',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-23T16:25:50.258Z'),
    updatedAt: new Date('2024-02-23T16:25:50.258Z'),
    treeItemId: '2ec6bba4-6dd0-4190-a015-b58e6ee4021c',
    name: '12346123445',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-21T20:51:41.137Z'),
    updatedAt: new Date('2024-02-21T20:51:41.137Z'),
    treeItemId: 'e4a9bc6a-cfb1-48ac-bc6e-1b6f135147c0',
    name: 'Chat Name 2',
    type: 'chat',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-20T19:47:03.738Z'),
    updatedAt: new Date('2024-02-20T19:47:03.738Z'),
    treeItemId: 'ce2422ea-b573-4c71-b569-fa2c79a5e50b',
    name: 'B matter with B name',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-20T19:44:31.642Z'),
    updatedAt: new Date('2024-02-20T19:44:31.642Z'),
    treeItemId: '148f72ac-b903-451c-88f2-ed1282c6acb1',
    name: 'Zmatter with Z name',
    matterId: 'A matter with A name',
    type: 'folder',
    fileCollectionId: 'whatever'
  },
  {
    createdAt: new Date('2024-02-15T18:35:41.269Z'),
    updatedAt: new Date('2024-02-15T18:35:41.269Z'),
    treeItemId: 'e82e81e6-68c8-4bd1-b67c-316dc4105d18',
    name: '123',
    matterId: 'Matter with a really REALLY long name, im mean like REALLY long',
    type: 'folder',
    fileCollectionId: 'whatever'
  }
];
