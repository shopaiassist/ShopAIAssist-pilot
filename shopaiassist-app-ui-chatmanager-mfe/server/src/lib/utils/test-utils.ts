import express, { NextFunction } from 'express';
import { AppRequest } from 'server/src/@types/appRequest';

/**
 * Sample data representing chat items for testing sorting and other functionalities.
 */
export const testChatsData = [
  {
    type: 'chat',
    name: 'New name for this chat',
    treeItemId: 'e35607c7-be8f-4172-a3cd-ae7e39bc2b97',
    createdAt: '2024-04-25T18:23:45.704Z',
    updatedAt: '2024-04-25T18:23:45.704Z',
    uid: '8beab79b628b479cb9c40855777a511d',
    parentId: '8c1b78b3-a798-4557-871a-bff6c8ac804e'
  },
  {
    createdAt: '2024-03-01T17:00:33.564Z',
    updatedAt: '2024-03-01T17:00:33.564Z',
    treeItemId: 'e35607c7-be8f-4172-a3cd-ae7e39bc2b96',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'Chat Name 1',
    type: 'chat'
  },
  {
    createdAt: '2024-02-21T20:51:41.137Z',
    updatedAt: '2024-02-21T20:51:41.137Z',
    treeItemId: 'e4a9bc6a-cfb1-48ac-bc6e-1b6f135147c0',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'Chat Name 2',
    type: 'chat'
  },
  {
    createdAt: '2024-02-21T20:49:41.137Z',
    updatedAt: '2024-02-21T20:49:41.137Z',
    treeItemId: 'c807ff14-58da-4199-8832-da90252fa743',
    parentId: 'b5014a24-8f61-42eb-a24e-8f0b88fd18ab',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'Archived Chat',
    type: 'chat'
  }
];

/**
 * Sample data representing folder items for testing sorting and other functionalities.
 */
export const testFoldersData = [
  {
    createdAt: '2024-02-23T16:30:33.080Z',
    updatedAt: '2024-02-23T16:30:33.080Z',
    treeItemId: '8c1b78b3-a798-4557-871a-bff6c8ac804e',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'new2',
    type: 'folder'
  },
  {
    createdAt: '2024-02-23T16:30:17.078Z',
    updatedAt: '2024-02-23T16:30:17.078Z',
    treeItemId: 'b39f37a3-6781-4445-8573-77658d9ee7ae',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'new1',
    type: 'folder'
  },
  {
    createdAt: '2024-02-23T16:26:17.254Z',
    updatedAt: '2024-02-23T16:26:17.254Z',
    treeItemId: '9bc3c583-a6e4-41cc-92f1-27e1be3ba4f6',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'newnewenewnew',
    type: 'folder',
    parentId: '8c1b78b3-a798-4557-871a-bff6c8ac804e'
  },
  {
    createdAt: '2024-02-23T16:20:00.254Z',
    updatedAt: '2024-02-23T16:20:00.254Z',
    treeItemId: 'b5014a24-8f61-42eb-a24e-8f0b88fd18ab',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'archived folder',
    type: 'folder',
    isArchived: true
  }
];

/**
 * Sample data for testing sorting chat and folder items by date.
 */
export const chatsAndFoldersByDate = [
  {
    createdAt: '2024-03-01T17:00:33.564Z',
    updatedAt: '2024-03-01T17:00:33.564Z',
    treeItemId: 'e35607c7-be8f-4172-a3cd-ae7e39bc2b96',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'Chat Name 1',
    type: 'chat'
  },
  {
    createdAt: '2024-02-23T16:30:33.080Z',
    updatedAt: '2024-02-23T16:30:33.080Z',
    treeItemId: '8c1b78b3-a798-4557-871a-bff6c8ac804e',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'new2',
    type: 'folder'
  },
  {
    createdAt: '2024-02-23T16:30:17.078Z',
    updatedAt: '2024-02-23T16:30:17.078Z',
    treeItemId: 'b39f37a3-6781-4445-8573-77658d9ee7ae',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'new1',
    type: 'folder'
  },
  {
    createdAt: '2024-02-21T20:51:41.137Z',
    updatedAt: '2024-02-21T20:51:41.137Z',
    treeItemId: 'e4a9bc6a-cfb1-48ac-bc6e-1b6f135147c0',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'Chat Name 2',
    type: 'chat'
  }
];

/**
 * Sample data for testing sorting chat and folder items by name.
 */
export const chatsAndFoldersByName = [
  {
    createdAt: '2024-03-01T17:00:33.564Z',
    updatedAt: '2024-03-01T17:00:33.564Z',
    treeItemId: 'e35607c7-be8f-4172-a3cd-ae7e39bc2b96',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'Chat Name 1',
    type: 'chat'
  },
  {
    createdAt: '2024-02-21T20:51:41.137Z',
    updatedAt: '2024-02-21T20:51:41.137Z',
    treeItemId: 'e4a9bc6a-cfb1-48ac-bc6e-1b6f135147c0',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'Chat Name 2',
    type: 'chat'
  },
  {
    createdAt: '2024-02-23T16:30:17.078Z',
    updatedAt: '2024-02-23T16:30:17.078Z',
    treeItemId: 'b39f37a3-6781-4445-8573-77658d9ee7ae',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'new1',
    type: 'folder'
  },
  {
    createdAt: '2024-02-23T16:30:33.080Z',
    updatedAt: '2024-02-23T16:30:33.080Z',
    treeItemId: '8c1b78b3-a798-4557-871a-bff6c8ac804e',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'new2',
    type: 'folder'
  }
];

/**
 * Sample data for testing sorting chat and folder items by type.
 */
export const chatsAndFoldersByType = [
  {
    createdAt: '2024-03-01T17:00:33.564Z',
    updatedAt: '2024-03-01T17:00:33.564Z',
    treeItemId: 'e35607c7-be8f-4172-a3cd-ae7e39bc2b96',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'Chat Name 1',
    type: 'chat'
  },
  {
    createdAt: '2024-02-21T20:51:41.137Z',
    updatedAt: '2024-02-21T20:51:41.137Z',
    treeItemId: 'e4a9bc6a-cfb1-48ac-bc6e-1b6f135147c0',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'Chat Name 2',
    type: 'chat'
  },
  {
    createdAt: '2024-02-23T16:30:33.080Z',
    updatedAt: '2024-02-23T16:30:33.080Z',
    treeItemId: '8c1b78b3-a798-4557-871a-bff6c8ac804e',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'new2',
    type: 'folder'
  },
  {
    createdAt: '2024-02-23T16:30:17.078Z',
    updatedAt: '2024-02-23T16:30:17.078Z',
    treeItemId: 'b39f37a3-6781-4445-8573-77658d9ee7ae',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'new1',
    type: 'folder'
  }
];

/**
 * Sample data for testing sorting chat and folder items by type.
 */
export const archivedChatsAndFoldersByType = [
  {
    createdAt: '2024-02-23T16:20:00.254Z',
    updatedAt: '2024-02-23T16:20:00.254Z',
    treeItemId: 'b5014a24-8f61-42eb-a24e-8f0b88fd18ab',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'archived folder',
    type: 'folder',
    isArchived: true
  }
];

/**
 * Sample data for testing sorting chat and folder items by type.
 */
export const archivedSubChatsAndFoldersByType = [
  {
    createdAt: '2024-02-21T20:49:41.137Z',
    updatedAt: '2024-02-21T20:49:41.137Z',
    treeItemId: 'c807ff14-58da-4199-8832-da90252fa743',
    parentId: 'b5014a24-8f61-42eb-a24e-8f0b88fd18ab',
    uid: '8beab79b628b479cb9c40855777a511d',
    name: 'Archived Chat',
    type: 'chat'
  }
];

export const requestHeaders = [
  'authorization',
  'op-orch 0702241037270OsnGa3dYt2HQ63uQhF3xaVG9i3ctHLqykE6E8KI1ydoGl05xQPMYPzpz44SBTWG1nxQryAKW7OmWwsZUK9z31yAWY3Q91b1puRZso9CW5n7MF9HXhnuwqNOaeODJw5_bTwg_Ky2ZdAhEI6p9GKSSBirRcRxU4veo-beuiyJ7OPXaRqxy5uV2kahg9dp4_P2PYeIbVcQV-p9wGy2mlekRfCNIp8aY97kjQOGKuQdvOkbIHm3d8b_IgT90rf1QQ3BoOSaH-gX5odIDYxz8eJAl83wgn98x3RxJuYkowEQ_VzyWFM4x4fCHu7E_dgML2Xr1UpR4I-PaY5JL6DPGkQr3QA'
];

const ONEPASS_ORCHESTRATION_TOKEN_TYPE = 'op-orch';
const UNAUTHORIZED_BODY = { message: 'No authorized user', code: 'no_authorized_user' };
const INVALID_TOKEN_BODY = { message: 'Invalid authorization token type', code: 'invalid_auth_token_type' };

/**
 * This is a mock of the security middleware from Hades for use in unit tests.
 */
export class MockSecurityMiddleware {
  constructor() {}

  /**
   * Requires the user to be authenticated and adds the logged in user profile
   * and orchestration token to the express request object
   *
   * @param req   An express request with the `user` and `orchestrationToken` property added to it
   * @param res   An express response
   * @param next  An express next function
   */
  public isAuthenticated = async (req: AppRequest, res: express.Response, next: NextFunction) => {
    const userMap = new Map();
    userMap.set(requestHeaders[1], {
      email: 'hunter.nenneau+localtest1@thomsonreuters.com',
      firstName: 'Hunter',
      lastName: 'Nenneau',
      id: '8beab79b628b479cb9c40855777a511d',
      username: 'hunter.nenneau+localtest1@thomsonreuters.com',
      registrationKey: ''
    });
    try {
      const { authorization } = req.headers;
      if (authorization) {
        const [type, token] = authorization.split(' ');
        if (type === ONEPASS_ORCHESTRATION_TOKEN_TYPE) {
          try {
            req.orchestrationToken = token;
            req.user = userMap.get(authorization);
            next();
          } catch (err: unknown) {
            next(err);
          }
        } else if (type) {
          res.status(400).send(INVALID_TOKEN_BODY);
        }
      } else {
        res.status(401).send(UNAUTHORIZED_BODY);
      }
    } catch (err) {
      next(err);
    }
  };
}
