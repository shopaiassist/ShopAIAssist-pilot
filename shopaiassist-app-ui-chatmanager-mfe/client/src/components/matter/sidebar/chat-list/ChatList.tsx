import ChatOrFolderItem from './ChatOrFolderItem';
import { useChatList } from '../../../../hooks/useChatList';

import './ChatList.scss';
import { useChatSidebarStore } from '../../../../store/useChatSidebarStore';
import { TreeItem, FolderItem } from '../../../../@types/sidebar';
import { useMatters } from '../../../../views/Context';
import { ModalType } from '../../../../App';
import useMFECommunication from '../../../../hooks/useMFECommunication';

interface ChatListProps {
  chatsAndFolders?: Array<TreeItem | FolderItem>;
  closeSidebar?: () => void;
}

/**
 * A component that renders a list of chat and folder items. It utilizes a custom hook to handle
 * the logic for sorting and selecting active items within the list.
 *
 * This component takes an optional `chatsAndFolders` array and renders a structured list
 * that groups chats and folders by a specified timeframe (if byDate sorting is selected). Each item in the list is displayed
 * using the `ChatOrFolderItem` component, which can handle selection and action management
 * for each individual item.
 *
 * @param  [chatsAndFolders] - An array containing chat or folder objects to be displayed.
 * @returns The rendered list of chat and folder items. If no items are provided,  a prompt to start adding items is displayed.
 *
 */
const ChatList = ({ chatsAndFolders, closeSidebar }: ChatListProps) => {
  const [sortedChatsAndFolders, activeItem, updateActiveItem] = useChatList(chatsAndFolders);
  const state = useChatSidebarStore();
  const { modalType } = useMatters();
  const [sendEvent] = useMFECommunication('chat_item_selected');
  return (
    <div className="chat-list-container">
      {chatsAndFolders?.length ? (
        <div className="chat-list">
          {state.sortType === 'by_date'
            ? sortedChatsAndFolders?.map((timeChunk, index) => (
                <div key={`${timeChunk.timeframe}-${index}`}>
                  <h4 className="timeframe-text">{timeChunk.timeframe}</h4>
                  {timeChunk.chats.map((treeItem) => (
                    <ChatOrFolderItem
                      treeItem={treeItem}
                      selected={activeItem?.treeItemId === treeItem.treeItemId}
                      onSelect={(item) => {
                        sendEvent({ message: 'chat_item_selected', body: activeItem });
                        updateActiveItem(item);
                        if (modalType === ModalType.Compact) {
                          closeSidebar && closeSidebar();
                        }
                      }}
                      key={`${treeItem.type}-${treeItem.treeItemId}`}
                    />
                  ))}
                </div>
              ))
            : chatsAndFolders.map((treeItem) => (
                <ChatOrFolderItem
                  treeItem={treeItem}
                  selected={activeItem?.treeItemId === treeItem.treeItemId}
                  onSelect={(item) => {
                    updateActiveItem(item);
                    if (modalType === ModalType.Compact) {
                      closeSidebar && closeSidebar();
                    }
                  }}
                  key={`${treeItem.type}-${treeItem.treeItemId}`}
                />
              ))}
        </div>
      ) : (
        <div className="empty-chat-list"> Chats will appear here</div>
      )}
    </div>
  );
};
export default ChatList;
