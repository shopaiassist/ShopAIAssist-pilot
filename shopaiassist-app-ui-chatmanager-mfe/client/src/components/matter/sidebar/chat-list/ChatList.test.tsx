import dayjs from 'dayjs';
import { snapshot } from '../../../../../jest-utils';
import ChatList from './ChatList';
import { mockTree } from '../../../../utils/test-utils/mockData';

const mockTreeData = mockTree;

const getMockedComponent = () => {
  return <ChatList chatsAndFolders={mockTreeData.slice(0, 5)} />;
};

describe('ChatList Tests', () => {
  describe('render', () => {
    // Run these tests as if today is 5/9/2024 so that the snapshot shows the chats grouped correctly by date.
    beforeEach(() => jest.useFakeTimers().setSystemTime(dayjs('2024-05-09T15:26:57.992Z').toDate()));
    afterEach(() => jest.useRealTimers());

    it('should render', async () => {
      snapshot(getMockedComponent());
    });
  });
});
