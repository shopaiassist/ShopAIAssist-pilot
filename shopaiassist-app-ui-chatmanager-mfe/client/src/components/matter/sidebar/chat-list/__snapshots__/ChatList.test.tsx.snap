// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatList Tests render should render 1`] = `
<div
  class="chat-list-container"
>
  <div
    class="chat-list"
  >
    <div>
      <div
        class="timeframe-text"
      >
        Past 7 Days
      </div>
      <div
        class="chat-or-folder-item-container"
      >
        <div
          class="selectable-item"
          data-testid="folder-9a4144e0-65ea-4f6c-8cd5-43a0a6d1e634-selectable"
        >
          <div
            appearance="light"
            class="chat-or-folder-icon"
            iconname="folder"
            size="16"
          />
          <span
            appearance="body-strong-md"
            id="chat-text-9a4144e0-65ea-4f6c-8cd5-43a0a6d1e634"
          >
            Test Folder For Created At 2
          </span>
        </div>
        <div
          class="chat-or-folder-menu"
        >
          <div
            class="selectable-item selectable-icon"
            data-testid="folder-9a4144e0-65ea-4f6c-8cd5-43a0a6d1e634-option-menu"
          >
            <div
              iconname="ellipsis-vertical"
              size="24"
            />
          </div>
        </div>
      </div>
      <div
        class="chat-or-folder-item-container"
      >
        <div
          class="selectable-item"
          data-testid="folder-5aa13d2b-3f04-4a91-a84b-bf214420b47c-selectable"
        >
          <div
            appearance="light"
            class="chat-or-folder-icon"
            iconname="folder"
            size="16"
          />
          <span
            appearance="body-strong-md"
            id="chat-text-5aa13d2b-3f04-4a91-a84b-bf214420b47c"
          >
            Test Folder For Created At 2
          </span>
        </div>
        <div
          class="chat-or-folder-menu"
        >
          <div
            class="selectable-item selectable-icon"
            data-testid="folder-5aa13d2b-3f04-4a91-a84b-bf214420b47c-option-menu"
          >
            <div
              iconname="ellipsis-vertical"
              size="24"
            />
          </div>
        </div>
      </div>
      <div
        class="chat-or-folder-item-container"
      >
        <div
          class="selectable-item"
          data-testid="chat-e35607c7-be8f-4172-a3cd-ae7e39bc2b96-selectable"
        >
          <div
            appearance="light"
            class="chat-or-folder-icon"
            iconname="message-lines"
            size="16"
          />
          <span
            appearance="body-strong-md"
            id="chat-text-e35607c7-be8f-4172-a3cd-ae7e39bc2b96"
          >
            A chat named desire
          </span>
        </div>
        <div
          class="chat-or-folder-menu"
        >
          <div
            class="selectable-item selectable-icon"
            data-testid="chat-e35607c7-be8f-4172-a3cd-ae7e39bc2b96-option-menu"
          >
            <div
              iconname="ellipsis-vertical"
              size="24"
            />
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="timeframe-text"
      >
        Past 30 Days
      </div>
      <div
        class="chat-or-folder-item-container"
      >
        <div
          class="selectable-item"
          data-testid="folder-4c484fb3-faf6-4fe8-9930-410772f141d7-selectable"
        >
          <div
            appearance="light"
            class="chat-or-folder-icon"
            iconname="folder"
            size="16"
          />
          <span
            appearance="body-strong-md"
            id="chat-text-4c484fb3-faf6-4fe8-9930-410772f141d7"
          >
            New Folder 100
          </span>
        </div>
        <div
          class="chat-or-folder-menu"
        >
          <div
            class="selectable-item selectable-icon"
            data-testid="folder-4c484fb3-faf6-4fe8-9930-410772f141d7-option-menu"
          >
            <div
              iconname="ellipsis-vertical"
              size="24"
            />
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="timeframe-text"
      >
        February 2024
      </div>
      <div
        class="chat-or-folder-item-container"
      >
        <div
          class="selectable-item"
          data-testid="folder-8c1b78b3-a798-4557-871a-bff6c8ac804e-selectable"
        >
          <div
            appearance="light"
            class="chat-or-folder-icon"
            iconname="folder"
            size="16"
          />
          <span
            appearance="body-strong-md"
            id="chat-text-8c1b78b3-a798-4557-871a-bff6c8ac804e"
          >
            new2
          </span>
        </div>
        <div
          class="chat-or-folder-menu"
        >
          <div
            class="selectable-item selectable-icon"
            data-testid="folder-8c1b78b3-a798-4557-871a-bff6c8ac804e-option-menu"
          >
            <div
              iconname="ellipsis-vertical"
              size="24"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
